explore/src/main/scala/explore/Routing.scala

```diff
@@ -52,135 +60,151 @@ object Routing:
-            programSummaries,
-            userPrefs,
-            routingInfo.focused,
-            model.rootModel.zoom(RootModel.searchingTarget),
-            model.rootModel.zoom(RootModel.expandedIds.andThen(ExpandedIds.asterismObsIds)),
+            model.rootModel.zoom(RootModel.vault).get,
+            programSummaries.model.zoom(ProgramSummaries.attachments),
+            programSummaries.model.get.obsAttachmentAssignments,
+            programSummaries.model.zoom(ProgramSummaries.observations),
+            userPreferences(model.rootModel).overviewTabLayout,
+            programSummaries.get.proposalIsAccepted,
             programSummaries.get.proposalIsSubmitted || model.userIsReadonlyCoi
           )
-        .toOption
+      .orEmpty
+
+  private def targetTab(page: Page, model: RootModelViews): VdomElement =
+    RoutingInfo
+      .from(page)
+      .map: routingInfo =>
+        withProgramSummaries(model): programSummaries =>
+          model.rootModel
+            .zoom(RootModel.userPreferences)
+            .mapValuePot: userPrefs =>
+              TargetTabContents(
+                routingInfo.programId,
+                model.rootModel.zoom(RootModel.userId).get,
+                programSummaries,
+                userPrefs,
+                routingInfo.focused,
+                model.rootModel.zoom(RootModel.searchingTarget),
+                model.rootModel.zoom(RootModel.expandedIds.andThen(ExpandedIds.asterismObsIds)),
+                programSummaries.get.proposalIsSubmitted || model.userIsReadonlyCoi
+              )
+            .toOption
+      .orEmpty
 
   private def obsTab(page: Page, model: RootModelViews): VdomElement =
-    val routingInfo = RoutingInfo.from(page)
-    withProgramSummaries(model): programSummaries =>
-      model.rootModel
-        .zoom(RootModel.userPreferences)
-        .mapValuePot: userPrefs =>
-          ObsTabContents(
-            model.rootModel.zoom(RootModel.vault).get,
+    RoutingInfo
+      .from(page)
+      .map: routingInfo =>
+        withProgramSummaries(model): programSummaries =>
+          model.rootModel
+            .zoom(RootModel.userPreferences)
+            .mapValuePot: userPrefs =>
+              ObsTabContents(
+                model.rootModel.zoom(RootModel.vault).get,
+                routingInfo.programId,
+                programSummaries,
+                userPrefs,
+                model.rootModel
+                  .zoom(RootModel.spectroscopyModes)
+                  .get
+                  .toOption
+                  .getOrElse(SpectroscopyModesMatrix.empty),
+                routingInfo.focused,
+                model.rootModel.zoom(RootModel.searchingTarget),
+                model.rootModel.zoom(RootModel.expandedIds.andThen(ExpandedIds.obsListGroupIds)),
+                programSummaries.get.proposalIsSubmitted || model.userIsReadonlyCoi
+              )
+            .toOption
+      .orEmpty
```

Comment:
I guess this is the case we can't parse the url, would it present an empty page then?

Reply:
It's the case where a `RoutingInfo` cannot be extracted from the `page`.

Up until now, all `Page`s produced a `RoutingInfo`, but now there are `Page`s that don't.

However, at the moment, we are only calling these methods with `Page`s from which `RoutingInfo` can always be extracted, and the `orEmpty` should never be hit. This is not future-proof tough.

We have some duplicate logic in the routing, we should revise things at some point and have better logic that doesn't require those `orEmpty`s, but it's not trivial.